Updated the ci to fail if lint failed

shabeebk opened the pull request:
Please fill in the fields below to submit a pull request.  The more information
that is provided, the better.

Fixes #1248

**Description of the changes being introduced by the pull request**:
The CI pipeline has been changed to run the lint tasks in the first and fail if the lint job fails.

**Please verify and check that the pull request fulfills the following
requirements**:

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

jku committed:
examples: Add skeleton API endpoints

Also use ThreadingHTTPServer: Using Chrome utterly breaks the non-threading
server.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
examples: Add skeleton API endpoints

Also use ThreadingHTTPServer: Using Chrome utterly breaks the non-threading
server.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
examples: Add further scaffolding for upload API

The API doesn't modify the repository yet but the data flow is there now.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
examples: Add further scaffolding for upload API

The API doesn't modify the repository yet but the data flow is there now.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
examples: Add uploader tool example

This tool works with the example repository: it can be used to
* Add a delegation (this is an unsafe API corresponding to e.g.
  project creation in PyPI)
* Submit new delegated role version (this requires using signing keys
  already submitted with the delegation)

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
examples: Add uploader tool example

This tool works with the example repository: it can be used to
* Add a delegation (this is an unsafe API corresponding to e.g.
  project creation in PyPI)
* Submit new delegated role version (this requires using signing keys
  already submitted with the delegation)

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
examples: Implement the upload API

uploader API has two POST endpoints

/api/delegation/<ROLE>
    Accepts new delegation keys for targetpath "<ROLE>/*" to role <ROLE>.
    This data is not signed in any way: In a real service this action would
    require some external authentication.

    POST content:
        { <KEYID>: <TUF KEY> }

/api/role/<ROLE>
    accepts uploads of new versions of <ROLE> metadata. The metadata
    must be correctly signed by the keys assigned to this delegation.

    POST content:
        TUF targets metadata as json

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
examples: Implement the upload API

uploader API has two POST endpoints

/api/delegation/<ROLE>
    Accepts new delegation keys for targetpath "<ROLE>/*" to role <ROLE>.
    This data is not signed in any way: In a real service this action would
    require some external authentication.

    POST content:
        { <KEYID>: <TUF KEY> }

/api/role/<ROLE>
    accepts uploads of new versions of <ROLE> metadata. The metadata
    must be correctly signed by the keys assigned to this delegation.

    POST content:
        TUF targets metadata as json

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
examples: Explain uploader tool in READMEs

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
examples: Explain uploader tool in READMEs

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
examples: Rename client example directory

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
examples: Rename client example directory

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

Commit:
Install pydocstyle for checking docstrings

Signed-off-by: Fridolin Pokorny <fridolin.pokorny@datadoghq.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

Commit:
Install pydocstyle for checking docstrings

Signed-off-by: Fridolin Pokorny <fridolin.pokorny@datadoghq.com>

dependabot[bot] committed:
build(deps): bump black from 21.12.0 to 23.1.0

Bumps [black](https://github.com/psf/black) from 22.12.0 to 23.1.0.
- [Release notes](https://github.com/psf/black/releases)
- [Changelog](https://github.com/psf/black/blob/main/CHANGES.md)
- [Commits](https://github.com/psf/black/compare/22.12.0...23.1.0)

---
updated-dependencies:
- dependency-name: black
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump black from 21.12.0 to 23.1.0

Bumps [black](https://github.com/psf/black) from 22.12.0 to 23.1.0.
- [Release notes](https://github.com/psf/black/releases)
- [Changelog](https://github.com/psf/black/blob/main/CHANGES.md)
- [Commits](https://github.com/psf/black/compare/22.12.0...23.1.0)

---
updated-dependencies:
- dependency-name: black
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
Whitespace changes to make new black linter happy

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
Whitespace changes to make new black linter happy

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
build: Move requirements file to a directory

We already have 6 files and I'm planning to add another one: maybe it's
time to move these out of the top level directory.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
build: Move requirements file to a directory

We already have 6 files and I'm planning to add another one: maybe it's
time to move these out of the top level directory.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
build: Split lint and test requirements

This way lint tool limitations don't prevent testing on older Python
versions.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
build: Split lint and test requirements

This way lint tool limitations don't prevent testing on older Python
versions.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
build: Bump isort from 5.11.4 to 5.12.0

New version requires python >=3.8 but that should be ok now with the
refactored requirements files.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
build: Bump isort from 5.11.4 to 5.12.0

New version requires python >=3.8 but that should be ok now with the
refactored requirements files.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
build: Handle GH release manually

Remove dependency on softprops/action-gh-release: instead do the GitHub
release steps using the GitHub API and github-script.

The only difference should be that release name is not "<tag>-rc" first:
instead the initial release is marked as draft in the API (and shows as
draft in the UI).

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
build: Handle GH release manually

Remove dependency on softprops/action-gh-release: instead do the GitHub
release steps using the GitHub API and github-script.

The only difference should be that release name is not "<tag>-rc" first:
instead the initial release is marked as draft in the API (and shows as
draft in the UI).

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

dependabot[bot] committed:
build(deps): bump pylint from 2.16.0 to 2.16.1

Bumps [pylint](https://github.com/PyCQA/pylint) from 2.16.0 to 2.16.1.
- [Release notes](https://github.com/PyCQA/pylint/releases)
- [Commits](https://github.com/PyCQA/pylint/compare/v2.16.0...v2.16.1)

---
updated-dependencies:
- dependency-name: pylint
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Updated the ci to fail if lint failed

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Updated the ci to fail if lint failed

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
build: shorten requirements file names

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
build: shorten requirements file names

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku commented:
There seems to be something wrong with the workflow. By the way, you can test CI changes in your own fork as well -- you just have to enable actions in the settings of your fork, then you can got the actions tab, choose the CI workflow, click "run workflow" and select "issue-1248" branch.

Commit:
Fix pydocstyle D209

D209: Multi-line docstring closing quotes should be on a separate line

Signed-off-by: Fridolin Pokorny <fridolin.pokorny@datadoghq.com>

Commit:
Fix pydocstyle D209

D209: Multi-line docstring closing quotes should be on a separate line

Signed-off-by: Fridolin Pokorny <fridolin.pokorny@datadoghq.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Updated the ci to continue if experimental job failed

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Updated the ci to continue if experimental job failed

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Updated the ci to continue if experimental job failed

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Updated the ci to continue if experimental job failed

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
updated the exclude list to fix yaml syntax

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
updated the exclude list to fix yaml syntax

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Added missing key failure-allowed

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Added missing key failure-allowed

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Updated the keys in matrix

Some of the keys in the matrix are being referenced in the upstream steps.

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Updated the keys in matrix

Some of the keys in the matrix are being referenced in the upstream steps.

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Removed the multiple linting jobs

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Removed the multiple linting jobs

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk commented:
@jku Thanks for reviewing it. I have fixed it now and ran successfully in my repository.

<img width="1440" alt="Screen Shot 2023-02-06 at 9 57 15 AM" src="https://user-images.githubusercontent.com/25004086/217048598-ddb11f1d-4510-4b0e-b2d7-e6657d6748e9.png">

jku commented:
I think I'll leave review for lukas (or I'll have to try to understand this better) but I will mention that the strange "waiting for status to be reported" is just because some checks are marked as required in project configuration: the check names have changed so the UI looks confused.

dependabot[bot] committed:
build(deps): bump mypy from 0.991 to 1.0.0

Bumps [mypy](https://github.com/python/mypy) from 0.991 to 1.0.0.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v0.991...v1.0.0)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump mypy from 0.991 to 1.0.0

Bumps [mypy](https://github.com/python/mypy) from 0.991 to 1.0.0.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v0.991...v1.0.0)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 2.2.1 to 2.2.2

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.2.1 to 2.2.2.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/3ebbd71c74ef574dbc558c82f70e52732c8b44fe...39d8d7e78f59cf6b40ac3b9fbebef0c753d7c9e5)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 2.2.1 to 2.2.2

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.2.1 to 2.2.2.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/3ebbd71c74ef574dbc558c82f70e52732c8b44fe...39d8d7e78f59cf6b40ac3b9fbebef0c753d7c9e5)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

lukpueh commented:
Thanks for the patch, @shabeebk! Have you tried if this works as [you intended](https://github.com/theupdateframework/python-tuf/issues/1248#issuecomment-1418285289)? Specifically:
> 1. Run the lint job as the first job in the matrix.

I don't think that your patch changes anything about the order in which jobs run. IIUC they still run in parallel depending on available resources.

dependabot[bot] committed:
build(deps): bump cryptography from 39.0.0 to 39.0.1 in /requirements

Bumps [cryptography](https://github.com/pyca/cryptography) from 39.0.0 to 39.0.1.
- [Release notes](https://github.com/pyca/cryptography/releases)
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/39.0.0...39.0.1)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump cryptography from 39.0.0 to 39.0.1 in /requirements

Bumps [cryptography](https://github.com/pyca/cryptography) from 39.0.0 to 39.0.1.
- [Release notes](https://github.com/pyca/cryptography/releases)
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/39.0.0...39.0.1)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
examples: Improve repository README

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
examples: Improve repository README

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
examples: Improve uploader docs/messages

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
examples: Improve uploader docs/messages

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
examples: Add missing link in repository README

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
examples: Add missing link in repository README

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
examples: More tweaks to uploader README

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
examples: More tweaks to uploader README

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
examples: Tweak download dir creation

Create target download dir when it's needed, not during "tofu".

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>

jku committed:
examples: Tweak download dir creation

Create target download dir when it's needed, not during "tofu".

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 2.2.2 to 2.2.3

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.2.2 to 2.2.3.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/39d8d7e78f59cf6b40ac3b9fbebef0c753d7c9e5...8775e868027fa230df8586bdf502bbd9b618a477)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 2.2.2 to 2.2.3

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.2.2 to 2.2.3.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/39d8d7e78f59cf6b40ac3b9fbebef0c753d7c9e5...8775e868027fa230df8586bdf502bbd9b618a477)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump requests from 2.28.1 to 2.28.2

Bumps [requests](https://github.com/psf/requests) from 2.28.1 to 2.28.2.
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.28.1...v2.28.2)

---
updated-dependencies:
- dependency-name: requests
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump requests from 2.28.1 to 2.28.2

Bumps [requests](https://github.com/psf/requests) from 2.28.1 to 2.28.2.
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.28.1...v2.28.2)

---
updated-dependencies:
- dependency-name: requests
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump charset-normalizer from 2.1.1 to 3.0.1

Bumps [charset-normalizer](https://github.com/Ousret/charset_normalizer) from 2.1.1 to 3.0.1.
- [Release notes](https://github.com/Ousret/charset_normalizer/releases)
- [Changelog](https://github.com/Ousret/charset_normalizer/blob/master/CHANGELOG.md)
- [Upgrade guide](https://github.com/Ousret/charset_normalizer/blob/master/UPGRADE.md)
- [Commits](https://github.com/Ousret/charset_normalizer/compare/2.1.1...3.0.1)

---
updated-dependencies:
- dependency-name: charset-normalizer
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump charset-normalizer from 2.1.1 to 3.0.1

Bumps [charset-normalizer](https://github.com/Ousret/charset_normalizer) from 2.1.1 to 3.0.1.
- [Release notes](https://github.com/Ousret/charset_normalizer/releases)
- [Changelog](https://github.com/Ousret/charset_normalizer/blob/master/CHANGELOG.md)
- [Upgrade guide](https://github.com/Ousret/charset_normalizer/blob/master/UPGRADE.md)
- [Commits](https://github.com/Ousret/charset_normalizer/compare/2.1.1...3.0.1)

---
updated-dependencies:
- dependency-name: charset-normalizer
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 2.2.3 to 2.2.4

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.2.3 to 2.2.4.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/8775e868027fa230df8586bdf502bbd9b618a477...17573ee1cc1b9d061760f3a006fc4aac4f944fd5)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 2.2.3 to 2.2.4

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.2.3 to 2.2.4.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/8775e868027fa230df8586bdf502bbd9b618a477...17573ee1cc1b9d061760f3a006fc4aac4f944fd5)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump pylint from 2.16.1 to 2.16.2

Bumps [pylint](https://github.com/PyCQA/pylint) from 2.16.1 to 2.16.2.
- [Release notes](https://github.com/PyCQA/pylint/releases)
- [Commits](https://github.com/PyCQA/pylint/compare/v2.16.1...v2.16.2)

---
updated-dependencies:
- dependency-name: pylint
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump pylint from 2.16.1 to 2.16.2

Bumps [pylint](https://github.com/PyCQA/pylint) from 2.16.1 to 2.16.2.
- [Release notes](https://github.com/PyCQA/pylint/releases)
- [Commits](https://github.com/PyCQA/pylint/compare/v2.16.1...v2.16.2)

---
updated-dependencies:
- dependency-name: pylint
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump mypy from 1.0.0 to 1.0.1

Bumps [mypy](https://github.com/python/mypy) from 1.0.0 to 1.0.1.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v1.0.0...v1.0.1)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump mypy from 1.0.0 to 1.0.1

Bumps [mypy](https://github.com/python/mypy) from 1.0.0 to 1.0.1.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v1.0.0...v1.0.1)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump coverage from 7.1.0 to 7.2.0

Bumps [coverage](https://github.com/nedbat/coveragepy) from 7.1.0 to 7.2.0.
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.1.0...7.2.0)

---
updated-dependencies:
- dependency-name: coverage
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump coverage from 7.1.0 to 7.2.0

Bumps [coverage](https://github.com/nedbat/coveragepy) from 7.1.0 to 7.2.0.
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.1.0...7.2.0)

---
updated-dependencies:
- dependency-name: coverage
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
repository: Make snapshot/timetamp helpers non-abstract

targets_infos() and snapshot_info() are helpers used by snapshot and
timestamp. Some Repository implementations do not need
snapshot/timestamp (think e.g. a signing tool that never modifies online
roles), so the helpers should not be required.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

jku committed:
repository: Make snapshot/timetamp helpers non-abstract

targets_infos() and snapshot_info() are helpers used by snapshot and
timestamp. Some Repository implementations do not need
snapshot/timestamp (think e.g. a signing tool that never modifies online
roles), so the helpers should not be required.

Signed-off-by: Jussi Kukkonen <jkukkonen@google.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump coverage from 7.2.0 to 7.2.1

Bumps [coverage](https://github.com/nedbat/coveragepy) from 7.2.0 to 7.2.1.
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.2.0...7.2.1)

---
updated-dependencies:
- dependency-name: coverage
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump coverage from 7.2.0 to 7.2.1

Bumps [coverage](https://github.com/nedbat/coveragepy) from 7.2.0 to 7.2.1.
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.2.0...7.2.1)

---
updated-dependencies:
- dependency-name: coverage
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 2.2.4 to 2.2.5

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.2.4 to 2.2.5.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/17573ee1cc1b9d061760f3a006fc4aac4f944fd5...32dc499307d133bb5085bae78498c0ac2cf762d5)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 2.2.4 to 2.2.5

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.2.4 to 2.2.5.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/17573ee1cc1b9d061760f3a006fc4aac4f944fd5...32dc499307d133bb5085bae78498c0ac2cf762d5)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
moved lint as a seperate job

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
moved lint as a seperate job

shabeebk commented:
Thanks, @lukpueh @jku for reviewing and for the feedback. I have updated the job to run lint separately before starting the test job. Added a snap of the job tree below.

The job is written in such a way that it will proceed only if lint succeeds.

eg: https://github.com/shabeebk/python-tuf/actions/runs/4301754458

![Screenshot from 2023-03-01 01-03-45](https://user-images.githubusercontent.com/25004086/222092928-39d941b1-0aea-4e2b-b3d2-8105ce4d142b.png)

lukpueh commented on .github/workflows/_test.yml:
```diff
@@ -5,24 +5,54 @@ on:
 jobs:
+  lint-test:
+    name: Lint Test
+    strategy:
+      matrix:
+        python-version: [3.x]
+
+    env:
+      TOXENV: lint
```
```suggestion
```
No need for this. You only run one toxenv on one python-version. You can specify them directly below.

shabeebk replied:
That's right. Updated to follow your thoughts.

lukpueh commented on .github/workflows/_test.yml:
```diff
@@ -5,24 +5,54 @@ on:
   tests:
     name: Tests
+    needs: lint-test
+    continue-on-error: ${{ matrix.failure-allowed }}
```
I don't think you need continue-on-error anymore, or do you?

shabeebk replied:
Yes, that is to keep the job running if anyone of them fails.

lukpueh commented on .github/workflows/_test.yml:
```diff
@@ -5,24 +5,54 @@ on:
     strategy:
-      fail-fast: false
       # Run regular TUF tests on each OS/Python combination, plus special tests
       # (sslib master) and linters on Linux/Python3.x only.
```
Comment is not fully accurate anymore (the "and linters" part)

shabeebk replied:
Updated. Thanks for pointing that out. :smiley:

lukpueh commented on .github/workflows/_test.yml:
```diff
@@ -75,7 +103,7 @@ jobs:
     # TODO: Replace always() with a 'at least one job succeeded' expression
     if: always()
-    needs: tests
+    needs: [lint-test, tests]
```
Shouldn't this work as is by transitivity? If "coveralls-fin" needs "tests" and "tests" needs "lint-test", then 
"coveralls-fin" needs "lint-test" without specifying it explicitly, right?

shabeebk replied:
Yes, you are right. Just fixed it.

lukpueh reviewed (changes requested):
Yeah, I think this is what we want. Many thanks for the continued efforts. I left a few minor comments, please address and we can merge this.

Oh, and please also sign-off all commits so that the DCO check passes.

shabeebk committed:
Refactored lint job

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Refactored lint job

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Updated the comment in workflow file

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Updated the comment in workflow file

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Merge branch 'develop' into issue-1248

Signed-off-by: Shabeeb Khalid <convey2shabeeb@gmail.com>

shabeebk committed:
Merge branch 'issue-1248' of github.com:shabeebk/python-tuf into issue-1248

shabeebk commented:
Closed in favour of a cleaner PR #2317
